packages/x-data-grid/src/internals/utils/computeSlots.ts

```diff
@@ -16,6 +18,7 @@ export function computeSlots<SlotComponents extends object>({
     const k = key as keyof typeof overrides;
 
     if (overrides[k] !== undefined) {
+      checkSlot(key, overrides[k] as any);
```

Comment:
Would consider wrapping the callsite instead of the function body to tree-shake it completely
```suggestion
      if (process.NODE_ENV !== 'production') {
        checkSlot(key, overrides[k] as any);
      }
```